drivers/src/tla2528.cpp

```diff
@@ -0,0 +1,137 @@
+#include <libhal-util/bit.hpp>
+#include <libhal-util/i2c.hpp>
+#include <libhal/error.hpp>
+#include <libhal/units.hpp>
+#include <tla2528.hpp>
+
+namespace sjsu::drivers {
+
+tla2528::tla2528(hal::i2c& p_i2c, hal::byte p_i2c_address)
+  : m_bus(p_i2c)
+{
+  m_i2c_address = p_i2c_address;
+  // TODO: reset command
```

Comment:
```suggestion
tla2528::tla2528(hal::i2c& p_i2c, hal::byte p_i2c_address)
  : m_bus(p_i2c),
  m_i2c_address(p_i2c_address)
{
  
  // TODO: reset command

```

ALSO TODO: Add a means to check if the device is available on the bus. If the device has an ID register you can look for it here. And throw an exception if the ID is wrong.

Reply:
Wouldn't a reset i2c call detect a lack of device through a NACK? Also, I couldn't find a register or operation like you were describing for checking a device ID.

Reply:
Kinda. Having an ID register is helpful since multiple devices can have the same I2C address and acknowledge the same payload as the reset command. We wouldn't actually know if we spoke to the appropriate device. But if it doesn't exist, then this should be good enough.  Can you add an issue to track this TODO. The format for todos should be:

```C++
// TODO(#12): Add reset command here. Will also double as detection of the i2c device.
```